contracts/staking/test/unit_tests/zrx_vault_test.ts

```diff
@@ -0,0 +1,404 @@
+        expect(eventArgs[0].staker).to.equal(staker);
+        expect(eventArgs[0].amount).to.bignumber.equal(amount);
+
+        const newVaultBalance = await zrxVault.balanceOf.callAsync(staker);
+        const newTokenBalance = await erc20Wrapper.getBalanceAsync(staker, zrxAssetData);
+        const [expectedVaultBalance, expectedTokenBalance] =
+            transferType === ZrxTransfer.Deposit
+                ? [initialVaultBalance.plus(amount), initialTokenBalance.minus(amount)]
+                : [initialVaultBalance.minus(amount), initialTokenBalance.plus(amount)];
+        expect(newVaultBalance).to.bignumber.equal(expectedVaultBalance);
+        expect(newTokenBalance).to.bignumber.equal(expectedTokenBalance);
+    }
+
+    describe('Normal operation', () => {
+        describe('Setting proxies', () => {
+            async function verifyStakingProxySetAsync(
+                receipt: TransactionReceiptWithDecodedLogs,
+                newAddress: string,
+            ): Promise<void> {
+                const eventArgs = filterLogsToArguments<ZrxVaultStakingProxySetEventArgs>(
+                    receipt.logs,
+                    'StakingProxySet',
+                );
+                expect(eventArgs.length).to.equal(1);
+                expect(eventArgs[0].stakingProxyAddress).to.equal(newAddress);
+                const actualAddress = await zrxVault.stakingProxyAddress.callAsync();
+                expect(actualAddress).to.equal(newAddress);
+            }
+
+            it('Owner can set the ZRX proxy', async () => {
+                const newAddress = nonOwnerAddresses[0];
+                const receipt = await zrxVault.setZrxProxy.awaitTransactionSuccessAsync(newAddress, {
+                    from: owner,
+                });
+                const eventArgs = filterLogsToArguments<ZrxVaultZrxProxySetEventArgs>(receipt.logs, 'ZrxProxySet');
+                expect(eventArgs.length).to.equal(1);
+                expect(eventArgs[0].zrxProxyAddress).to.equal(newAddress);
+            });
+            it('Non-owner address cannot set the ZRX proxy', async () => {
+                const [notOwner, newAddress] = nonOwnerAddresses;
+                const tx = zrxVault.setZrxProxy.awaitTransactionSuccessAsync(newAddress, {
+                    from: notOwner,
+                });
+                const expectedError = new OwnableRevertErrors.OnlyOwnerError(notOwner, owner);
+                expect(tx).to.revertWith(expectedError);
+            });
+            it('Owner can set the staking proxy', async () => {
+                const newAddress = nonOwnerAddresses[0];
+                const receipt = await zrxVault.setStakingProxy.awaitTransactionSuccessAsync(newAddress, {
+                    from: owner,
+                });
+                await verifyStakingProxySetAsync(receipt, newAddress);
+            });
+            it('Authorized address can set the staking proxy', async () => {
```

Comment:
If we make `setZrxProxy` authorized, then we should add tests like this for that function.